Adding a node labeling utility script

KalyanChakravarthyK opened the pull request:
*Issue #, if available:*

*Description of changes:*

Adding a node labeling utility script that makes it easier to identify node types within a Kubernetes environment. This is particularly useful in heterogeneous cluster environments like the hybrid-hyperpod-eks configuration.


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

KalyanChakravarthyK committed:
Create node-label-by-type.sh

Adding a node labeling utility script that makes it easier to identify node types within a Kubernetes environment. This is particularly useful in heterogeneous cluster environments like the hybrid-hyperpod-eks configuration

KalyanChakravarthyK committed:
Update install-kubeps1.sh

KalyanChakravarthyK commented:
@iankouls-aws can you review this please

iankouls-aws commented:
Hi @KalyanChakravarthyKodela,
Thanks for submitting a PR!
There actually already is a label for the instance type it it is included in the nodes-list.sh (alias `kgn`) script.
Could you please try:
`kubectl get nodes -L node.kubernetes.io/instance-type`
or `kgn` within the container.
Does this serve your needs?
If not, please clarify. Also, would you mind setting the alias to `nlt` instead of `nl-type` to make it shorter and easier to type?
I understand that from the full instance type (e.g. ml.g5.8xlarge), you are extracting the middle section (e.g. g5), a.k.a as instance family. The entire string is considered the instance-type. The `ml` prefix indicates that this instance is owned by SageMaker. `g` is the instance series, `5` is the generation, and `8xlarge` is the instance size. The entire string is called instance-type. We need to refer to the part you extract as instance family - could you correct that in your code? The alias will then become `nlf` and the utility script `node-label-family.sh` 
Please refer to this documentation: https://docs.aws.amazon.com/ec2/latest/instancetypes/instance-type-names.html

KalyanChakravarthyK committed:
Update and rename node-label-by-type.sh to node-label-family.sh

Renamed the script from node-label-by-type.sh to node-label-family.sh to correctly reflect that we're labeling by instance family rather than instance type
Changed the label from node.hyperpod.aws/type to node.hyperpod.aws/family throughout the script to use the correct terminology
Changed the variable name from TYPE_PREFIX to INSTANCE_FAMILY to better reflect what the value represents

KalyanChakravarthyK committed:
Update install-kubeps1.sh

using nlf instead of nl-type

KalyanChakravarthyK commented:
Thank you for reviewing my PR. I've made the requested changes:

1. Renamed the script to `node-label-family.sh` to correctly reflect that it's labeling by instance family rather than instance type
2. Changed the label format to `node.hyperpod.aws/family` throughout the script 
3. Updated all terminology in the code and documentation to use "instance family" instead of "type prefix"
4. Changed the suggested alias to `nlf` as recommended

While I understand that the existing `kubectl get nodes -L node.kubernetes.io/instance-type` command (or `kgn`) displays the full instance type, the purpose of this script is different. It creates Kubernetes labels that make it easier to select nodes by their family (g5, p4d, etc.) rather than by their full instance type. This enables more flexible node selection for workloads that need specific hardware families but don't care about the exact size.

For example, with these labels, users can deploy workloads to any g5 instance regardless of size using a nodeSelector like:
```yaml
nodeSelector:
  node.hyperpod.aws/family: g5
```

This is particularly useful in heterogeneous clusters where you want to target a specific hardware family but be flexible about exact instance sizes.

Thank you for your guidance and open source contributions.

iankouls-aws commented:
The changes look good. Thank you @KalyanChakravarthyKodela for your thoughtful contribution!
